runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,758 @@
+>lua
+    vim.keymap.del('n', '<Leader>ex1')
+    vim.keymap.del({'n', 'c'}, '<Leader>ex2', {buffer = true})
+<
+------------------------------------------------------------------------------
+See also:
+• `vim.api.`|nvim_get_keymap()|     - return all global mapping
+• `vim.api.`|nvim_buf_get_keymap()| - return all mappings for buffer
+
+==============================================================================
+Autocommands                                            *lua-guide-autocommands*
+
+An |autocommand| is a Vim command or a Lua function that is automatically
+executed whenever one or more |events| are triggered, e.g., when a file is
+read or written, or when a window is created. These are accessible from Lua
+through the Neovim API.
+
+------------------------------------------------------------------------------
+Creating autocommands                             *lua-guide-autocommand-create*
+
+Autocommands are created using `vim.api.`|nvim_create_autocmd()|, which takes
+two mandatory arguments:
+• {event}: a string or table of strings containing the event(s) which should
+           trigger the command or function.
+• {opts}:  a table with keys that control what should happen when the event(s)
+           are triggered.
+
+The most important options are:
+
+• `pattern`:  A string or table of strings containing the |autocmd-pattern|.
+            Note: Environment variable like `$HOME` and `~` are not automatically
+            expanded; you need to explicitly use `vim.fn.`|expand()| for this.
+• `command`:  A string containing a Vim command.
+• `callback`: A Lua function.
+
+You must specify one and only one of `command` and `callback`. If `pattern` is
+omitted, it defaults to `pattern = '*'`.
+Examples:
+>lua
+    vim.api.nvim_create_autocmd({"BufEnter", "BufWinEnter"}, {
+      pattern = {"*.c", "*.h"},
+      command = "echo 'Entering a C or C++ file'",
+    })
+
+    -- Same autocommand written with a Lua function instead
+    vim.api.nvim_create_autocmd({"BufEnter", "BufWinEnter"}, {
+      pattern = {"*.c", "*.h"},
+      callback = function() print("Entering a C or C++ file") end,
+    })
+
+    -- User event triggered by MyPlugin
+    vim.api.nvim_create_autocmd("User", {
+      pattern = "MyPlugin",
+      callback = function() print("My Plugin Works!") end,
+    })
+<
+
+Neovim will always call a Lua function with a single table containing information
+about the triggered autocommand. The most useful keys are
+• `match`: a string that matched the `pattern` (see |<amatch>|)
+• `buf`:   the number of the buffer the event was triggered in (see |<abuf>|)
+• `file`:  the file name of the buffer the event was triggered in (see |<afile>|)
+• `data`:  a table with other relevant data that is passed for some events
+
+For example, this allows you to set buffer-local mappings for some filetypes:
+>lua
+    vim.api.nvim.create_autocmd("FileType", {
+      pattern = "lua",
```

Comment:
nit: hmm, the function is valid as is but should this be `*.lua` instead? Maybe you do want to match only files named `lua`, but perhaps we should use a more recognizable static file name such as `Dockerfile` or `CMakeLists.txt`?

Reply:
It's a `FileType` autocommand, so the pattern _is_ the filetype, not a path glob.

Reply:
Do you expect me to know everything about vim /s